File: check_categories.py
import os
from supabase import create_client
from dotenv import load_dotenv

load_dotenv()

SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_KEY = os.getenv("SUPABASE_KEY")

supabase = create_client(SUPABASE_URL, SUPABASE_KEY)

# Get all categories
categories = supabase.table("categories").select("*").order("display_order").execute()

print("\n=== CATEGORIES ===")
for cat in categories.data:
    print(f"ID: {cat['id']}")
    print(f"Name: {cat['name']}")
    print(f"Slug: {cat['slug']}")
    print(f"Display Order: {cat['display_order']}")
    print("-" * 50)

# Get Mahalle Yardımı related categories and subcategories
print("\n=== MAHALLE YARDIMI RELATED ===")
mahalle_cats = [c for c in categories.data if 'Mahalle' in c['name'] or 'Nachbarschaft' in c['name']]
for cat in mahalle_cats:
    print(f"\nCategory: {cat['name']} (ID: {cat['id']})")
    subs = supabase.table("subcategories").select("*").eq("category_id", cat['id']).order("display_order").execute()
    for sub in subs.data:
        print(f"  - Subcategory: {sub['name']} (ID: {sub['id']})")

# Check Eğlence, Hobi & Mahalle category
print("\n=== EĞLENCE, HOBI & MAHALLE ===")
eglence_cats = [c for c in categories.data if 'Eğlence' in c['name'] or 'Hobi' in c['name'] or 'Freizeit' in c['name']]
for cat in eglence_cats:
    print(f"\nCategory: {cat['name']} (ID: {cat['id']})")
    subs = supabase.table("subcategories").select("*").eq("category_id", cat['id']).order("display_order").execute()
    for sub in subs.data:
        print(f"  - Subcategory: {sub['name']}")

File: app/frontend/check_categories.js
require('dotenv').config({ path: '../../.env' });
const { createClient } = require('@supabase/supabase-js');

const supabaseUrl = process.env.REACT_APP_SUPABASE_URL;
const supabaseKey = process.env.REACT_APP_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseKey) {
    console.error("Missing Supabase credentials");
    process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function checkCategories() {
    const { data: categories, error } = await supabase
        .from('categories')
        .select('id, name');

    if (error) {
        console.error("Error fetching categories:", error);
        return;
    }

    console.log("Categories:");
    for (const cat of categories) {
        console.log(`- "${cat.name}"`);
        if (cat.name.includes('Musik')) {
            const { data: subs, error: subError } = await supabase
                .from('subcategories')
                .select('name')
                .eq('category_id', cat.id);
            if (subs) {
                console.log("  Subcategories:");
                subs.forEach(s => console.log(`    - "${s.name}"`));
            }
        }
    }
}

checkCategories();
